app/controllers/buyers_controller.rb

```diff
@@ -0,0 +1,47 @@
+class BuyersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :sold_out
+  before_action :this_is_saller
+
+  def index
+    @product = Product.find(params[:product_id])
```

Comment:
```ruby
@product = Product.find(params[:product_id])
```
indexとcreateアクションに重複した記述が存在するためbefore_actionでまとめましょう。

Reply:
```suggestion
  def set_product
    @product = Product.find(params[:product_id])
  end
```
重複していた記述をbefore_actionにまとめました。indexとcreate、sold_out、this_is_sallerの４箇所修正しました。